### PR TITLE
Feat/card expansion

### LINKED ISSUE #1240: [Feature Request] Card Reveal Panel for Displaying More Info for a Card
There seems to be pretty good coverage of cards but the ability to easily show a revealed panel is missing.   This will allow the user to:

* Configure some settings for a card
* Provide extra information for a card without having to browse somewhere else

I don't think this is an edge case, it is potentially useful for any card.

This is demonstrated in the Card Reveal Section here: http://materializecss.com/cards.html

Demo Video: https://db.tt/mNyIizI1QP

| Default | Open |
| ------------- | ------------- |
| image  | image |

### PR DESCRIPTION
**Pull request by Renato66**
## Description
New component for hidden card text

## Motivation and Context
closes #1240 
## How Has This Been Tested?
* unit tests

## Markup
```vue
<v-flex xs12 sm6>
  <v-card>
    <v-card-media
      src="/static/doc-images/cards/sunshine.jpg"
      height="200px"
    >
    </v-card-media>
    <v-card-title primary-title>
      <div>
        <div class="headline">Top western road trips</div>
        <span class="grey--text">1,000 miles of wonder</span>
      </div>
    </v-card-title>
    <v-card-actions>
      <v-btn flat>Share</v-btn>
      <v-btn flat color="purple">Explore</v-btn>
      <v-spacer></v-spacer>
      <v-btn icon @click.native="cover = !cover">
        <v-icon>more_vert</v-icon>
      </v-btn>
    </v-card-actions>
    <v-card-expansion v-show="cover" cover>
      <v-container pa-0>
        <v-layout row>
          <v-flex>
            <div class="headline">Top western road trips</div>
            <span class="grey--text">1,000 miles of wonder</span>
          </v-flex>
          <v-flex class="text-xs-right">
            <v-btn icon @click.native="cover = false">
              <v-icon>close</v-icon>
            </v-btn>
          </v-flex>
        </v-layout>
        <v-layout row>
          <v-flex xs12>
            I'm a thing. But, like most politicians, he promised more than he could deliver. You won't have time for sleeping, soldier, not with all the bed making you'll be doing. Then we'll go with that data file! Hey, you add a one and two zeros to that or we walk! You're going to do his laundry? I've got to find a way to escape.
          </v-flex>
        </v-layout>
      </v-container>
    </v-card-expansion>
  </v-card>
</v-flex>
```
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/components/VCard/VCardExpansion.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  render (h) {
+    const children = []
+    if (typeof this.$slots.default === 'undefined') consoleWarn('no text was provided', this)
+
+    children.push(this.$slots.default)
+
+    const staticClass = 'v-card__expansion ' + (this.cover ? 'v-card__expansion--cover' : 'v-card__expansion--dropdown')
+    const expansionStyle = h('div', {
+      'class': staticClass
```
Comment (johnleider) [inline src/components/VCard/VCardExpansion.js]: staticClass is a valid property

```
const expansionStyle = h('div', {
  staticClass,
  ...
}
```

File: src/components/VCard/VCardExpansion.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  props: {
+    cover: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  render (h) {
+    const children = []
+    if (typeof this.$slots.default === 'undefined') consoleWarn('no text was provided', this)
```
Comment (johnleider) [inline src/components/VCard/VCardExpansion.js]: remove this

File: src/components/VCard/VCardExpansion.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { VExpandTransition, VSlideYReverseTransition } from '../transitions'
+
+import { consoleWarn } from '../../util/console'
+
+export default {
+  name: 'v-card-expansion',
```
Comment (johnleider) [inline src/components/VCard/VCardExpansion.js]: this should be a functional component

File: src/components/VCard/VCardExpansion.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      type: Boolean,
+      default: false
+    }
+  },
+
+  render (h) {
+    const children = []
+    if (typeof this.$slots.default === 'undefined') consoleWarn('no text was provided', this)
+
+    children.push(this.$slots.default)
```
Comment (johnleider) [inline src/components/VCard/VCardExpansion.js]: this isn't needed, just pass whatever the value is

File: src/stylus/components/_cards.styl [hunk 5]
```diff
@@ -40,7 +43,17 @@ theme(v-card, "v-card")
...
-
+  &__expansion
+    &--cover
+      position: absolute
+      width: 100%
+      height: 100%
+      overflow-y: auto
+      left: 0
+      top: 0
+      bottom: 0
```
Comment (johnleider) [inline src/stylus/components/_cards.styl]: add space below

File: test/unit/components/VCard/VCardExpansion.spec.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            staticClass: 'fake-text'
+          })
+        }]
+      }
+    })
+
+    expect(wrapper.html()).toMatchSnapshot()
+  })
+
+  it('should render with cover class', () => {
```
Comment (johnleider) [inline test/unit/components/VCard/VCardExpansion.spec.js]: remove this test

**Comment by nekosaur** [comment]
Do we care that the `cover` variant transitions from outside the card? It's easier to see if you add a background color to `v-card-expansion`.

**Comment by johnleider** [comment]
Closing due to inactivity.

**Pull request closed without merging**